### DISCUSSION TITLE
Can't build grblHAL/LPC176x - missing header.ldt

### DISCUSSION THREAD

**Discussion by dgangola** [grblHAL/core#484] [Q&A] [answered]
I'm trying to build grblHAL/LPC176x, to run on a Smoothieboard.
Not wanting to download the IDE, I started out compiling this with CMake.  It went well, right up to the point I realized I couldn't link th project, as there was no .ld file.  So, I downloaded the IDE.
I STILL can't link, as there seems to be a missing file: "header.ldt".
The IDE gives the following error:

> Error while creating linker script for project GRBL Driver LPC176x.
> Script GRBL_Driver_LPC176x_Debug_memory.ld. Template memory.ldt
> Template inclusion failed (for parameter value "header.ldt"):
> Template not found for name "header.ldt".
> The name was interpreted by this TemplateLoader: com.crt.advproject.linkertemplate.FMLinkerFileTemplateLoader@7d9380fe.
> 
> FTL stack trace ("~" means nesting-related):
> 	- Failed at: #include "header.ldt"  [in template "memory.ldt" at line 3, column 1]
> 

I'm new to the NXP IDE, and the managed link system.  I can't help think I'm missing something obvious.  "header.ldt" does not seem to be included with the IDE, I've searched the container it's running in, and the file is not there.  I searched the git commit history, and the file was never committed.  Was it just left out of the repo, or does everyone but me know where to find "header.ldt"?

Thanks,
   Dave

**Comment by terjeio** [comment, marked as answer]
It is part of the IDE, I found it here: _<install folder>\MCUXpressoIDE_11.3.1_5262\ide\Wizards\linker_

[linker.zip](https://github.com/grblHAL/core/files/14895892/linker.zip)

**Reply by dgangola** [reply]
Thanks for getting back to me so quickly.  And, thanks for the .zip, it saved me having to download the Windows version.
Those files are not included with the Linux version.  The documentation says they will be (I read it after the fact),  but, they're not. No Wizards/linker folder.  I did a search through all the .zip files in the installation, and those files aren't in any of them.
